Use common strings in Nest config flow

chrillebile opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Use common strings in nest config flow as described in #40578.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: #40578 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

chrillebile committed:
Add common strings

chrillebile committed:
Fix typo

springstan commented on homeassistant/components/nest/strings.json:
```diff
@@ -9,19 +9,19 @@
-      "no_flows": "You need to configure Nest before being able to authenticate with it. [Please read the instructions](https://www.home-assistant.io/components/nest/).",
-      "authorize_url_timeout": "Timeout generating authorize url.",
+      "already_setup": "[%key:common::config_flow::abort::single_instance_allowed%]",
+      "no_flows": "[%key:common::config_flow::abort::oauth2_missing_configuration%]",
```
Change the key here as well:
```suggestion
      "missing_configuration": "[%key:common::config_flow::abort::oauth2_missing_configuration%]",
```

springstan commented on homeassistant/components/nest/strings.json:
```diff
@@ -9,19 +9,19 @@
-      "already_setup": "You can only configure a single Nest account.",
-      "no_flows": "You need to configure Nest before being able to authenticate with it. [Please read the instructions](https://www.home-assistant.io/components/nest/).",
-      "authorize_url_timeout": "Timeout generating authorize url.",
+      "already_setup": "[%key:common::config_flow::abort::single_instance_allowed%]",
```
Here too
```suggestion
      "single_instance_allowed": "[%key:common::config_flow::abort::single_instance_allowed%]",
```

springstan reviewed (commented):
`oauth2_pick_implementation` could be used as well

chrillebile committed:
Change keys

- Change no_flows to missing_configuration
- Change already_setup to single_instance_allowed

chrillebile commented:
I don't understand why the tests failed, could someone explain?

springstan commented:
> I don't understand why the tests failed, could someone explain?

You have to adjust the tests for nest as well e.g. [test_config_flow.py](https://github.com/home-assistant/core/blob/dev/tests/components/nest/test_config_flow.py)

chrillebile commented:
> > I don't understand why the tests failed, could someone explain?
> 
> You have to adjust the tests for nest as well e.g. [test_config_flow.py](https://github.com/home-assistant/core/blob/dev/tests/components/nest/test_config_flow.py)

aah, sorry I forgot! Thanks!

chrillebile committed:
Fix tests

Fixes tests for new common string key names

springstan reviewed (changes requested):
The files in ./translations do not have to be adjusted, please revert your changes for these files.

chrillebile committed:
Revert ./translations

chrillebile committed:
Add invalid_pin to common string

springstan commented on homeassistant/components/nest/strings.json:
```diff
@@ -2,26 +2,26 @@
     "step": {
       "init": {
-        "title": "Authentication Provider",
+        "title": "[%key:common::config_flow::title::oauth2_pick_implementation%]",
         "description": "Pick via which authentication provider you want to authenticate with Nest.",
```
I would switch these two lines, because the title does not include a description of what to do next.
```suggestion
        "title": "Authentication Provider",
        "description": "[%key:common::config_flow::title::oauth2_pick_implementation%]",
```

chrillebile committed:
Switch authentication title and description

springstan reviewed (approved):
LGTM :v:
